pkg/util/testutil/timeout.go

```diff
@@ -14,6 +14,19 @@ import (
 // is deadlocked.
 func AssertTrueBeforeTimeout(t *testing.T, frequency, timeout time.Duration, condition func() bool) {
 	t.Helper()
+	internalTrueBeforeTimeout(t, frequency, timeout, condition)
+}
+
+// RequireTrueBeforeTimeout is the same as AssertTrueBeforeTimeout, but it calls
+// t.failNow() if the condition function times out.
+func RequireTrueBeforeTimeout(t *testing.T, frequency, timeout time.Duration, condition func() bool) {
+	t.Helper()
+	if !internalTrueBeforeTimeout(t, frequency, timeout, condition) {
+		t.FailNow()
+	}
+}
+
+func internalTrueBeforeTimeout(t *testing.T, frequency, timeout time.Duration, condition func() bool) bool {
```

Comment:
ok, then let's call this `assert...` to better reflect what it does:

```suggestion
func assertTrueBeforeTimeout(t *testing.T, frequency, timeout time.Duration, condition func() bool) bool {
```